homeassistant/components/enocean/__init__.py

```diff
@@ -1,93 +1,59 @@
 )
 
-SIGNAL_RECEIVE_MESSAGE = "enocean.receive_message"
-SIGNAL_SEND_MESSAGE = "enocean.send_message"
-
 
-def setup(hass, config):
+async def async_setup(hass, config):
     """Set up the EnOcean component."""
-    serial_dev = config[DOMAIN].get(CONF_DEVICE)
-    dongle = EnOceanDongle(hass, serial_dev)
-    hass.data[DATA_ENOCEAN] = dongle
-
-    return True
-
-
-class EnOceanDongle:
-    """Representation of an EnOcean dongle."""
-
-    def __init__(self, hass, ser):
-        """Initialize the EnOcean dongle."""
-
-        self.__communicator = SerialCommunicator(port=ser, callback=self.callback)
-        self.__communicator.start()
-        self.hass = hass
-        self.hass.helpers.dispatcher.dispatcher_connect(
-            SIGNAL_SEND_MESSAGE, self._send_message_callback
+    # support for text-based configuration (legacy)
+    if DOMAIN not in config:
+        return True
+
+    configured_dongle = {
+        entry.data[CONF_DEVICE] for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+    if configured_dongle:
+        # We can only have one dongle. If there is already one in the config,
+        # there is no need to import the yaml based config.
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
         )
+    )
 
-    def _send_message_callback(self, command):
-        """Send a command through the EnOcean dongle."""
-        self.__communicator.send(command)
-
-    def callback(self, packet):
-        """Handle EnOcean device's callback.
-
-        This is the callback function called by python-enocan whenever there
-        is an incoming packet.
-        """
-
-        if isinstance(packet, RadioPacket):
-            _LOGGER.debug("Received radio packet: %s", packet)
-            self.hass.helpers.dispatcher.dispatcher_send(SIGNAL_RECEIVE_MESSAGE, packet)
-
-
-class EnOceanDevice(Entity):
-    """Parent class for all devices associated with the EnOcean component."""
-
-    def __init__(self, dev_id, dev_name="EnOcean device"):
-        """Initialize the device."""
-        self.dev_id = dev_id
-        self.dev_name = dev_name
-
-    async def async_added_to_hass(self):
-        """Register callbacks."""
-        self.async_on_remove(
-            self.hass.helpers.dispatcher.async_dispatcher_connect(
-                SIGNAL_RECEIVE_MESSAGE, self._message_received_callback
-            )
-        )
+    return True
 
-    def _message_received_callback(self, packet):
-        """Handle incoming packets."""
 
-        if packet.sender_int == combine_hex(self.dev_id):
-            self.value_changed(packet)
+async def async_setup_entry(
+    hass: core.HomeAssistant, config_entry: config_entries.ConfigEntry
+):
+    """Set up an EnOcean dongle for the given entry."""
+    default_enocean_data = {}
+    enocean_data = hass.data.setdefault(DATA_ENOCEAN, default_enocean_data)
```

Comment:
```
Python 3.8.2 (default, Feb 26 2020, 02:56:10) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.13.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: data = {}                                                                                                                                                               

In [2]: test = data.setdefault("special_data", {})                                                                                                                              

In [3]: test                                                                                                                                                                    
Out[3]: {}

In [4]: data                                                                                                                                                                    
Out[4]: {'special_data': {}}
```

Reply:
Yes, I did the same test here, with the same result (but with a python 3.7.7 interpreter).
As I mentionned, it also only happens when creating a new integration, not when setting up an already configured integration.
I am currently upgrading to python 3.8.2, hoping it will help.

Reply:
It does work perfectly with python 3.8.2.
Use of the literal commited.